### PR TITLE
Create MsgRequestData and MsgReportData struct

### LINKED ISSUE #489: Create MsgRequestData and MsgReportData struct

### PR DESCRIPTION
**Pull request by taobun**
fixed #489 

Remove old msg and comment un-useable code

### CODE REVIEW

File: chain/x/zoracle/internal/types/msgs.go [hunk 1]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// MsgRequestData is a new version of request message.
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: MsgRequestData is a message for requesting a new data request to an existing oracle script.

File: chain/x/zoracle/internal/types/msgs.go [hunk 2]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+type MsgRequestData struct {
+	OracleScriptID           int64          `json:"oracleScriptID"`
+	Calldata                 []byte         `json:"calldata"`
+	RequestedValidatorCount  int64          `json:"requestedValidatorCount"`
+	SufficientValidatorCount int64          `json:"sufficientValidatorCount"`
+	Expiration               int64          `json:"expiration"`
+	Sender                   sdk.AccAddress `json:"sender"`
+}
+
+// NewMsgRequestData is a constructor function for MsgRequestData
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: NewMsgRequestData creates a new NewMsgRequestData instance.

File: chain/x/zoracle/internal/types/msgs.go [hunk 3]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+		OracleScriptID:           oracleScriptID,
+		Calldata:                 calldata,
+		RequestedValidatorCount:  requestedValidatorCount,
+		SufficientValidatorCount: sufficientValidatorCount,
+		Expiration:               expiration,
+		Sender:                   sender,
+	}
+}
+
+// Route should return the name of the module
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: Let's just do this for everything. "Route implements the sdk.Msg interface for MsgRequestData." so we don't need to worry about what to type.

File: chain/x/zoracle/internal/types/msgs.go [hunk 4]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+func (msg MsgRequestData) Route() string { return RouterKey }
+
+// Type should return the action
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: ditto.

File: chain/x/zoracle/internal/types/msgs.go [hunk 5]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+func (msg MsgRequestData) Type() string { return "request" }
+
+// ValidateBasic runs stateless checks on the message
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: ditto.

File: chain/x/zoracle/internal/types/msgs.go [hunk 6]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+	if msg.RequestedValidatorCount < msg.SufficientValidatorCount {
+		return sdk.ErrUnknownRequest("Request validator count must be greater than sufficient validator")
+	}
+	if msg.Expiration <= 0 {
+		return sdk.ErrUnknownRequest("Expiration period must be greater than zero")
+	}
+	return nil
+}
+
+// GetSigners defines whose signature is required
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: ditto

File: chain/x/zoracle/internal/types/msgs.go [hunk 7]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+func (msg MsgRequestData) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// GetSignBytes encodes the message for signing
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: ditto

File: chain/x/zoracle/internal/types/msgs.go [hunk 8]
```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
...
+
+// Route should return the name of the module
+func (msg MsgRequestData) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgRequestData) Type() string { return "request" }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgRequestData) ValidateBasic() sdk.Error {
+	if msg.Sender.Empty() {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go:47]: I will have a ticket to revisit all error messages later. For now it's low priority.

File: chain/x/zoracle/internal/types/msgs.go [hunk 9]
```diff
@@ -111,6 +177,57 @@ func (msg MsgReport) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// MsgReportData defines a new version Report message
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: MsgReport is a message sent by each of the block validators to respond to a data request.

File: chain/x/zoracle/internal/types/msgs.go [hunk 10]
```diff
@@ -111,6 +177,57 @@ func (msg MsgReport) GetSignBytes() []byte {
...
+type MsgReportData struct {
+	RequestID int64          `json:"requestID"`
+	Data      []ExternalData `json:"data"`
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: Let's call this `DataSet` so it feels like there's more than one.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/types/msgs.go (hunk 8 above)
Comment (taobun) [reply chain/x/zoracle/internal/types/msgs.go:47]: Sir yes sir

File: chain/x/zoracle/internal/types/msgs.go [hunk 11]
```diff
@@ -81,13 +147,13 @@ func NewMsgReport(
 	}
 }
 
-// Route should return the name of the module
+// Route implements the sdk.Msg interface for MsgRequestData.
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: For MsgReport

File: chain/x/zoracle/internal/types/msgs.go [hunk 12]
```diff
@@ -30,13 +30,13 @@ func NewMsgRequest(
...
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: For MsgRequest

File: chain/x/zoracle/internal/types/msgs.go [hunk 13]
```diff
@@ -131,13 +248,13 @@ func NewMsgStoreCode(
...
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go]: For MsgStoreCode

File: chain/x/zoracle/internal/types/msgs.go [hunk 14]
```diff
@@ -198,12 +315,12 @@ func (msg MsgDeleteCode) ValidateBasic() sdk.Error {
 	return nil
 }
 
-// GetSigners defines whose signature is required
+// GetSigners implements the sdk.Msg interface for MsgRequestData.
 func (msg MsgDeleteCode) GetSigners() []sdk.AccAddress {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/msgs.go:215]: For MsgDeleteCode

File: chain/x/zoracle/internal/types/report.go [hunk 15]
```diff
@@ -39,3 +39,17 @@ func NewValidatorReport(
 		Validator:  valAddress,
 	}
 }
+
+// ExternalData is a struct that capsulized external data id and raw data
+type ExternalData struct {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/report.go:44]: ExternalData encapsulates a raw data report for an external data source from a block validator.

File: chain/x/zoracle/internal/types/report.go [hunk 16]
```diff
@@ -39,3 +39,17 @@ func NewValidatorReport(
...
+	ExternalDataID int64  `json:"externalDataID"`
+	Data           []byte `json:"data"`
+}
+
+// NewExternalData is a constructor of ExternalData
+func NewExternalData(externalDataID int64, data []byte) ExternalData {
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/report.go:50]: NewExternalData creates a new ExternalData instance.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]
Looks good.

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Comment by evilpeach** [comment]
This PR is a break feature from Alpha-net branch

**Pull request merged**